litestar/contrib/pydantic/pydantic_schema_plugin.py

```diff
@@ -322,3 +355,34 @@ def for_pydantic_model(cls, field_definition: FieldDefinition, schema_creator: S
                 )
             ),
         )
+
+    @classmethod
+    def get_required_fields(
+        cls,
+        property_fields: dict[str, FieldDefinition],
+        exclude: PydanticFieldsList = None,
+        exclude_defaults: bool = False,
+        exclude_none: bool = False,
+        exclude_unset: bool = False,
+        include: PydanticFieldsList = None,
+    ) -> list[str]:
+        required = []
+        exclude = exclude or set()
+        include = include or []
+
+        for prop in property_fields.values():
+            name = prop.name
+
+            if any(
+                [
+                    name in exclude,
+                    exclude_none and (prop.is_optional or prop.is_none_type),
+                    (exclude_defaults or exclude_unset) and prop.has_default,
```

Comment:
I'm still a bit hesitant regarding this. This makes sense for the schema we generate for responses, but it doesn't make sense for the schema we generate for request bodies. Since we don't differentiate between whether the model is used as a request or response body, this may be problematic.